pkg/bpf/collection.go

```diff
@@ -41,6 +48,130 @@ func LoadCollectionSpec(path string) (*ebpf.CollectionSpec, error) {
 	return spec, nil
 }
 
+func removeUnreachableTailcalls(spec *ebpf.CollectionSpec) error {
+	type TailCall struct {
+		referenced bool
+		visited    bool
+		spec       *ebpf.ProgramSpec
+	}
+
+	entrypoints := make([]*ebpf.ProgramSpec, 0)
+	tailcalls := make(map[uint32]*TailCall)
+
+	for _, prog := range spec.Programs {
+		var id, slot uint32
+		// Consider any program that doesn't follow the tailcall naming convention
+		// x/y to be an entrypoint.
+		if _, err := fmt.Sscanf(prog.SectionName, "%d/%v", &id, &slot); err != nil {
+			log.Debugf("section '%s' / prog '%s', not a tailcall, considering an entrypoint", prog.SectionName, prog.Name)
+			entrypoints = append(entrypoints, prog)
+			continue
+		}
+
+		const (
+			cilium_calls_map = 2
```

Comment:
Can you add a warning to `bpf/lib/ids.h` and also make that owned by `sig-loader`?